pkg/operator/csi/csidrivercontrollerservicecontroller/helpers.go

```diff
@@ -63,3 +65,26 @@ func WithSecretHashAnnotationHook(
 		return nil
 	}
 }
+
+// WithReplicasHook sets the deployment.Spec.Replicas field according to the number
+// of available nodes. If the number of available nodes is bigger than one, then the
+// number of replicas will be two. The number of nodes is determined by the node
+// selector specified in the field deployment.Spec.Templates.NodeSelector.
+// When node ports or hostNetwork are used, maxSurge=0 should be set in the
+// Deployment RollingUpdate strategy to prevent the new pod from getting stuck
+// waiting for a node with free ports.
+func WithReplicasHook(nodeLister corev1listers.NodeLister) DeploymentHookFunc {
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
```

Comment:
Having a second thought about `WithReplicasHook`. Will we have an operator that does **not** want the hook? Would it make sense to have `${MASTER_REPLICAS}` in our yaml files instead?

Reply:
I did that initially, but I moved it to a hook mostly because it's easier to unit test it and because I wanted to enable this code path in a few operators first. 

I think it makes sense to promote it to a placeholder, although I don't think the hook is too bad as well. Can we do this in another PR?

Reply:
I may have found another reason a hook is better than a placeholder.

It's a chicken and egg issue. I need to replace the `${MASTER_REPLICAS}` string **before** parsing the manifest file (otherwise the parsing fails). However, I'd need to parse the manifest file to get the node selector **before** replacing the string.

A simple solution would be to drop the `${MASTER_REPLICAS}` placeholder and set the replicas regardless, but in that case a hook sounds better IMO.